sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -64,7 +64,20 @@ static TableChange setProperty(String property, String value) {
    * @return a TableChange for the addition
    */
   static TableChange removeProperty(String property) {
```

Comment:
Can I directly update this API or no?

Reply:
Yea we need to keep backward compatibility

Reply:
Another way is to do the work in Spark analyzer: if the key doesn't exist and `ifExists = true`, remove this table change.

Then we don't need to change API.

Reply:
Yes, the analyzer way seems better. But the rule will handle the `ifExists = false` case and error out if the properties do not exist, instead of removing the table changes for the `ifExists = true` case; catalog implementation may want to log nonexistent properties to inform the user, for example.